scripts/scr_marine_struct/scr_marine_struct.gml

```diff
@@ -0,0 +1,371 @@
+		static increase_max_health = function(increase){
+			max_health += (increase*(1+((constitution - 20)*0.05))); //calculate the effect of health buffs
+		}
+		static hp = function(){ 
+			return obj_ini.hp[company,marine_number] //return current health
+		};
+		//update_marine health
+       static update_health = function(new_health){
+            obj_ini.hp[company,marine_number] = new_health;
+	   };
+	   update_health(max_health); //set marine health to max
+	   
+		static weapon_one = function(){ 
+			return obj_ini.wep1[company,marine_number];
+		};
+       static update_weapon_one = function(new_weapon){
+            obj_ini.wep1[company,marine_number] = new_weapon;
+	   };
+		static weapon_two = function(){ 
+			return obj_ini.wep1[company,marine_number];
+		};
+       static update_weapon_two = function(new_weapon){
+            obj_ini.w[company,marine_number] = new_weapon;
+	   };
+		static armour = function(){ 
+			return obj_ini.armour[company,marine_number];
+		};	   
+       static update_armour = function(new_armour){
+            obj_ini.armour[company,marine_number] = new_armour;
+			get_unit_size(); //every time armour is changed see if the marines size has changed
+	   };	
+		static corruption = function(){ 
+			return obj_ini.chaos[company,marine_number];
+		};	   
+       static update_corruption = function(new_corruption){
+            obj_ini.chaos[company,marine_number] = new_corruption;
+	   };	
+		static specials = function(){ 
+			return obj_ini.spe[company,marine_number];
+		};	   
+       static update_specials = function(new_specials){
+            obj_ini.spe[company,marine_number] = new_specials;
+	   };
+	   	static mobility_item = function(){ 
+			return obj_ini.mobi[company,marine_number];
+		};	   
+       static update_mobility_item = function(new_mobility_item){
+            obj_ini.mobi[company,marine_number] = new_mobility_item;
+			get_unit_size(); //every time mobility_item is changed see if the marines size has changed
+	   };	
+	   	static race = function(){ 
+			return obj_ini.race[company,marine_number];
+		};	
+		static role = function(){
+			return obj_ini.role[company,marine_number];
+		};
+		static update_role = function(new_role){
+			obj_ini.role[company,marine_number]= new_role;
+			array_append(role_history ,[obj_ini.wep1[company,marine_number], obj_controller.turn])
+		};
+		
+		static get_unit_size = function(){
+			var r = role();
+			var arm = armour();
+			var sz = 0;
+			sz = 1;
+			var bulky_armour = ["Terminator Armour", "Tartaros"]
+		    if (string_count("Dread",arm)>0) {sz+=5;} else if (array_contains(bulky_armour,arm)){sz +=1};
+			var mobi =  mobility_item();
+			if (mobi == "Jump Pack"){
+				sz++;
+			}
+			if (r == "Chapter Master"){sz++}
+			size =sz;
+			return size
+		};
+		
+		static marine_location = function(){
+			var location_id,location_name;
+			var location_type = obj_ini.wid[company,marine_number];
+			if ( location_type > 0){ //if marine is on planet
```

Comment:
`location_type` seems like a good candidate for converting to an enum

Reply:
done anyway